content/blog/the-8-part-guide-to-better-code-review-checklists/index.md

```diff
@@ -0,0 +1,124 @@
+
+# 1. Pull Request Etiquette ✅
+
+Start with the basics. Is the pull request you are looking at *actually* ready for review? To determine this, here are some key principles that should apply to all code reviews:
+
+* **Frequent** — To make pull requests more manageable, submit them little and often. If a pull request covers multiple days with no good reason, it should be split into smaller chunks.
+* **Bitesize** — If a pull request is too large, it's difficult to understand. It should be broken down as much as possible.
+* **Well documented** — Does the pull request have enough comments to make it easy to review?
+* **Follows the Single Responsibility Principle** — Code should only ever solve one problem. If a piece of code affects a lot of the rest of the code in the project, it's difficult to review.
+
+If these conditions are not met, return the code to the contributor to be improved or separated into different pull requests.
+
+This might seem initially harsh or counter-productive, but over several weeks your team's outputs and productivity will improve.
+
+# 2. Styling 🎨
+
+Set some basic styling rules for your team and ensure these are followed. Consistent code styling is essential to ensure future developers can understand code easily and work productively.
+
+Some basic rules:
+* Are variable names sensible and consistently capitalized?
+* Are there sufficiently descriptive comments throughout the code as required?
+* Are formatting preferences followed? For example, tabs or spaces, curly brace on the same line or new line, 80 character width or 120 character width?
+
+To save time, you could check these rules are followed with an automatic code linter.
+
+Again, if these standards are not met, stop checking the code and return it to the contributor for review.
+
+# 3. Security 🔐
+
+Set security standards for your project and check these are being followed religiously. The rules to be checked will vary depending on your project and organization, but some best practices are:
+
+* Run your project through vulnerability scanning solutions, like LGTM
+* Don't hardcode credentials for testing or include secrets in the repository
+* Don't disclose too much information in error messages, which might give hints to an attacker
+* Ensure any database queries are parameterized
+* Don't trust user or client input blindly — your client or web application can be modified. For example, URL parameters that access resources should be checked.
+
+If you identify security issues in a code review process, stop and speak to the contributor. It might be a sign of a more serious issue in the project or a lack of training, both of which will need follow on interventions.
+
+# 4. Performance 🏎
+
+Is the project as performant as it could be, or are these obvious optimizations that might improve performance?
```

Comment:
Shout-out to [Chrome Lighthouse](https://developers.google.com/web/tools/lighthouse) (also called [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)) would be good in this section, as well as [DebugBear](https://www.debugbear.com) for frontend projects.